gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,595 @@
+import logging
+import random
+import numpy
+
+import numpy as np
+from gensim.scripts.glove2word2vec import glove2word2vec
+from gensim.test.utils import get_tmpfile
+from gensim.models import KeyedVectors
+from collections import Counter
+from custom_losses import rank_hinge_loss
+from custom_layers import TopKLayer
+from sklearn.preprocessing import normalize
+from custom_callbacks import ValidationCallback
+from gensim import utils
+
+try:
+    import keras.backend as K
+    from keras import optimizers
+    from keras.losses import hinge
+    from keras.models import Model
+    from keras.layers import Input, Embedding, Dot, Dense, Reshape, Dropout
+    import tensorflow
+    tensorflow.set_random_seed(101010)
+    KERAS_AVAILABLE = True
+except ImportError:
+    KERAS_AVAILABLE = False
+
+random.seed(101010)
+numpy.random.seed(101010)
+logger = logging.getLogger(__name__)
+
+
+class DRMM_TKS(utils.SaveLoad):
+    """User friendly model for training on similarity learning data.
+    You only have to provide sentences in the data as a list of words.
+
+    Example Usage:
```

Comment:
example MUST be executable, for example https://github.com/RaRe-Technologies/gensim/blob/fd6f6eb71996cd36a6236070f33efdfb37dde342/gensim/models/word2vec.py#L724-L731

this is easy to check: `python -m doctest path/to/your/py/file`